src/steps/command.ts

```diff
@@ -230,11 +230,16 @@ export class CommandStep extends LabeledStep {
     async toJson(
         opts: ToJsonSerializationOptions = { explicitDependencies: false },
     ): Promise<object> {
+        // Need to pull out one of env/retry to get around a weird Typescript v4.0 bug.
```

Comment:
great catch, I do think we should try and reproduce this and if we can repro it open an issue in the TS project.

Reply:
tried to reproduce with `4.0.2` in the playground:
```
async function fn4() {
  const a = () => Promise.resolve({a: 1 });
  console.log({
    ...(await a()),
    x: await (() => Promise.resolve({ x: 1 }))(),
    b: 1,
    y: await (() => Promise.resolve({ y: 1 }))(),
  })
}
```
but no luck so far.

Reply:
I also wasn't able to reproduce with [this playground](https://www.typescriptlang.org/play?#code/MYGwhgzhAECC0G8CwAoa7rAPYDsIBcAnAV2Hy0IAoAHYgIxAEthowAuaLOgKwFMyAlIgC+qDNDEZIATxwsA5r3yUhyNOIyElxQjmgAFQlgC2jCLwB0WiFhAA3XpXwALMxbACA3JPSiUf1FBIGAAhaF4AD3xeHAATGHg1cWw8IlJyKh8NWgZmaDoARg5YABos8RymFjoAJmKy9XRVcowIYmpeKgRWDgLoYS8sgMbWCFkFJRqVRBb0LXwdPSSNFYs1yjAAdzBGfGg2jsILRWUBAQaVlcKOLZ29lzdC46UVC8vxCI4ABjf39FqbttdtAHhALLVnqdfhphN4Rn5hikCPloABeaA4XibaAhShZTHY2CUbrXaAAckKZP653xWLgxPydXJtSpAwagxQdEhUwEFhcMUo1jRAD5MLgbCBLCAsPJKBSySVoNYzp4gA) using classes to somewhat mimic our use case. Maybe it also has to do with the `KeyValueImpl` and `RetryImpl` objects, or the type casts?

Reply:
I'd feel better if we knew why this was happening...